### ISSUE TITLE
Ubuntu compilation

### ISSUE THREAD

**Issue by dimiro1** [etexteditor/e#15]
I am getting the following error, when trying to compile e on ubuntu 8.10 with gcc 4.3.2

../external/out.release/lib/libwxwebkit.a(wxwebkit_static_WebView.o): In function `wxStringBase::wxStringBase(wxStringBase const&)':
WebView.cpp:(.text._ZN12wxStringBaseC2ERKS_[wxStringBase::wxStringBase(wxStringBase const&)]+0x48): undefined reference to`wxOnAssert(wchar_t const_, int, char const_, wchar_t const_, wchar_t const_)'
../external/out.release/lib/libwxwebkit.a(wxwebkit_static_WebView.o): In function `wxAutoBufferedPaintDC::TestWinStyle(wxWindow*)':
WebView.cpp:(.text._ZN21wxAutoBufferedPaintDC12TestWinStyleEP8wxWindow[wxAutoBufferedPaintDC::TestWinStyle(wxWindow*)]+0x4f): undefined reference to`wxOnAssert(wchar_t const_, int, char const_, wchar_t const_, wchar_t const_)'
../external/out.release/lib/libwebcore-wx.a(wx_webcore_static_StringWx.o): In function `String':
/home/claudemiro/projetos/e/external/webkit/WebCore/platform/text/wx/StringWx.cpp:80: undefined reference to`wxOnAssert(wchar_t const_, int, char const_, wchar_t const_, wchar_t const_)'
/home/claudemiro/projetos/e/external/webkit/WebCore/platform/text/wx/StringWx.cpp:80: undefined reference to `wxOnAssert(wchar_t const_, int, char const_, wchar_t const_, wchar_t const_)'
collect2: ld returned 1 exit status
make: **\* [e] Error 1

any suggestions?

**Comment by rndmerle** [comment]
Hi Dimiro.

You launched external/get-packages-ubuntu.sh right ?
Can you verify wx2.8-headers package is installed ?

**Comment by dimiro1** [comment]
Yes. The first thing i did was lauch external/get-packages-ubuntu.sh, The package wx2.8-headers is installed.

**Comment by dimiro1** [comment]
the error happen when ld is running.

**Comment by feniix** [comment]
Did you try using the latest wx package from the wx debian/ubuntu repository?
The afaik e uses 2.8.10 and even jaunty is lagging one revision behind.

**Comment by bluemoon** [comment]
you didnt run make DEBUG=1 , the code in webkit for some reason wants wxAssert which is only available in the debug build of wxwidgets, trying running it with "make DEBUG=1"

**Comment by jipumarino** [comment]
There is no chance of compiling it in release mode?

**Comment by bluemoon** [comment]
theres a few changes you have to make to webkit for it to compile in release mode

**Comment by bluemoon** [comment]
try looking at 891e3f1922585594346889ccf7a3ede916a4baeb

**Comment by jipumarino** [comment]
The octocat says "That page doesn't exist!" ;)

**Comment by bluemoon** [comment]
hah the page worked for me, lets see try http://github.com/bluemoon/e/commit/891e3f1922585594346889ccf7a3ede916a4baeb or bluemoon/e@891e3f1922585594346889ccf7a3ede916a4baeb
